File: CommunifyLibrary/NonPersistentModels/PageViewModels/ProfilePageViewModel.cs
using CommunifyLibrary.NonPersistentModels.ViewModels;

namespace CommunifyLibrary.NonPersistentModels.PageViewModels;
public class ProfilePageViewModel
{
    public UserInformationSummaryViewModel UserInformationSummary { get; set; }
    public ProfileStatsViewModel ProfileStats { get; set; }
    public ProfileStatusViewModel ProfileStatus { get; set; }
    public bool isSuccess { get; set; } = false;
}

File: web/src/models/pageViewModels/ProfilePageViewModel.ts
import { ProfileStatsViewModel } from "../viewModels/ProfileStatsViewModel"
import { ProfileStatusViewModel } from "../viewModels/ProfileStatusViewModel"
import { UserInformationSummaryViewModel } from "../viewModels/UserInformationSummaryViewModel"

export type ProfilePageViewModel = {
    userInformationSummary: UserInformationSummaryViewModel | null,
    profileStats: ProfileStatsViewModel | null,
    profileStatus: ProfileStatusViewModel | null,
    isSuccess: boolean
}